### ISSUE TITLE
Invalid Request.Url and CallingAppName values in RegisterMergeCallback

### ISSUE THREAD

**Issue by miyconst** [Starcounter/Starcounter#3700]
I have the following setup up:

- App Playground.
- App Subground.
- App Playground has a property `SubPage: { }`.
- App Playground fills the value via `Self.GET("/playground/sub");`.
- The `"/playground/sub"` is mapped to `UriMapping.MappingUriPrefix + "/playground-sub-page"`.
- App Subground also has a response mapped to `UriMapping.MappingUriPrefix + "/playground-sub-page"`.

If the Subground app was started first then:

```
Request.Url = "/playground/sub";
CallingAppName = "Playground";
```

If the Playground app was started first then:

```
Request.Url = "/subground/sub";
CallingAppName = "Subground";
```

I expect `Request.Url` to have the value provided to `Self.GET` and `CallingAppName` to have the name of the app which initialized the `Self.GET`. Regardless of application start order.

cc @alemoi, @chrhol

**Comment by miyconst** [comment]
Can be tested with my [Playground & Subground](https://github.com/miyconst/Playground/) apps.

**Comment by Mnosek** [comment]
I've just hit the same problem on following configuration:
- Starcounter v2.3.0.4846
- UserAdmin
- Launcher

Launcher settings page is mapped to `UriMapping.MappingUriPrefix + "/settings"`
UserAdmin settings page is mapped to `UriMapping.MappingUriPrefix + "/settings"`

The same as in Konstantin's case I expect `Request.Url and CallingAppName` to have a name of the app which initialised request. 

What is interesting, it occurs only if we map `Handle.GET` directly, when `Handle.GET` deals with partial and only partial is mapped, everything works as expected.

We've debugged it with a assistance of @chrhol, so he most likely knows more.

**Comment by chrhol** [comment]
Yeah we noticed this as well yesterday, and as @Mnosek writes it happens when there is a mapping to a handler called directly from client. 

I think that the problem is not in the mergecallback or merger itself, but stems from the wrong request being sent in to it. Instead of the one being called from outside, the request is the request from the last handler that was registered/mapped.

I will take a look and see if I can find what's happening. If I get stuck I might ask @alemoi for some help :smile:

**Comment by chrhol** [comment]
We decided to postpone this issue for some time and focus on getting 2.3 released first (which is scheduled quite soon).

**Comment by warpech** [comment]
From what was discussed between me, @alemoi and @chrhol, implementing https://github.com/Starcounter/Starcounter/issues/3966 will "fix" the problem.
